test/addons-napi/test_promise/test.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const common = require('../../common');
+const test_promise = require(`./build/${common.buildType}/test_promise`);
+const assert = require('assert');
+
+let native_promise, conclude_value;
+
+// Create a native promise and resolve it.
+native_promise = test_promise.napi_create_promise();
+conclude_value = 'Promise delivered';
+native_promise.promise
+  .then(
+    function(result) {
+      assert.strictEqual(result, conclude_value,
+                         'Promise was resolved as expected');
+    },
+    function(error) {
+      assert.ok(false,
+                'Promise was rejected when it should have been resolved: ' +
+                error + ': ' + JSON.stringify(error, null, 4));
+    });
+native_promise.resolve(conclude_value);
+
+// Make sure the native promise is recognized as a promise.
+assert.strictEqual(test_promise.napi_is_promise(native_promise.promise), true,
+                   'N-API correctly identifies a native promise as a promise');
+
+// Create a native promise and reject it.
+native_promise = test_promise.napi_create_promise();
+conclude_value = new Error('I broke my promise');
+native_promise.promise
+  .then(
+    function(result) {
+      assert.ok(false,
+                'Promise was resolved when it should have been rejected: ' +
+          result + ': ' + JSON.stringify(result, null, 4));
+    },
```

Comment:
If you want to be lazy, you can just do `common.mustNotCall()` instead of the `assert.ok(false)` calllback